packages/rum-core/src/domain/action/getActionNameFromElement.ts

```diff
@@ -288,12 +279,15 @@ function getTextualContentWithTreeWalker(
   return text.replace(/\s+/g, ' ').trim()
 
   function rejectInvisibleOrMaskedElementsFilter(node: Node) {
+    const nodeSelfPrivacyLevel = getNodePrivacyLevel(node, nodePrivacyLevel, nodePrivacyLevelCache)
+    const shouldRejectMaskedNode =
+      privacyEnabledActionName && nodeSelfPrivacyLevel && shouldMaskNode(node, nodeSelfPrivacyLevel)
```

Comment:
It seems like it would be less complicated to just early return here:
```suggestion
    if (privacyEnabledActionName && nodeSelfPrivacyLevel && shouldMaskNode(node, nodeSelfPrivacyLevel)) {
      return NodeFilter.FILTER_REJECT
    }
```